core/vm/lfvm/interpreter_test.go

```diff
@@ -18,6 +21,392 @@ type example struct {
 	function uint32 // The identifier of the function in the contract to be called
 }
 
+const MAX_STACK_SIZE int = 1024
+const GAS_START uint64 = 1 << 32
+
+var (
+	HASH_0    = common.Hash{0}
+	HASH_1    = common.BigToHash(big.NewInt(1))
+	HASH_2    = common.BigToHash(big.NewInt(2))
+	ADDRESS_0 = common.Address{0}
+)
+
+func getEmptyContext() context {
+	code := make([]Instruction, 0)
+	data := make([]byte, 0)
+	return getContext(code, data, 0, nil, GAS_START, false, false)
+}
+
+func getContext(code Code, data []byte, stackPtr int, stateDB vm.StateDB, gas uint64, isBerlin bool, isLondon bool) context {
+
+	// Create a dummy contract
+	addr := vm.AccountRef{}
+	contract := vm.NewContract(addr, addr, big.NewInt(0), gas)
+
+	// Create execution context.
+	ctxt := context{
+		code:     code,
+		data:     data,
+		callsize: *uint256.NewInt(uint64(len(data))),
+		stack:    NewStack(),
+		memory:   NewMemory(),
+		readOnly: false,
+		contract: contract,
+		stateDB:  stateDB,
+		isBerlin: isBerlin,
+		isLondon: isLondon,
+		evm:      &vm.EVM{StateDB: stateDB},
+	}
+
+	// Set start conditions
+	ctxt.pc = 0
+	ctxt.status = RUNNING
+	ctxt.stack.stack_ptr = stackPtr
+
+	return ctxt
+}
+
+// Test UseGas function and correct status after running out of gas
+func TestGasFunc(t *testing.T) {
+	ctx := getEmptyContext()
+	ctx.contract.Gas = 100
+	ok := ctx.UseGas(10)
+	if !ok {
+		t.Errorf("expected not failed useGas function, got failed")
+	}
+	if ctx.contract.Gas != 90 {
+		t.Errorf("expected gas of contract in context is 90, got %d", ctx.contract.Gas)
+	}
+	ok = ctx.UseGas(100)
+	if ok {
+		t.Errorf("expected failed useGas function, got ok")
+	}
+	if ctx.contract.Gas != 90 {
+		t.Errorf("expected gas of contract in context is 90 also after failing, got %d", ctx.contract.Gas)
+	}
+	if ctx.status != OUT_OF_GAS {
+		t.Errorf("expected OUT_OF_GAS status 6, got %d", ctx.status)
+	}
+}
+
+type OpcodeTest struct {
+	name        string
+	code        []Instruction
+	stackPtrPos int
+	argData     uint16
+	endStatus   Status
+	isBerlin    bool
+	isLondon    bool
```

Comment:
I understand, but the problem is, that chainconfig is implemented in a way, that if the block is newer than berlin and london fork, then it returns true for both. And some conditions in the code has to have both set to true to correctly indicate and calculate gas for blocks after london fork

Reply:
Maybe bitmask would be better for this

Reply:
Thanks for clarifying that london and berlin may be reasonably true at the same time and should be tested.

A bitmask would be nice, but this can also be done the next time a version jump is introduced.